common/beerocks/bwl/dummy/base_wlan_hal_dummy.cpp

```diff
@@ -133,7 +140,44 @@ bool base_wlan_hal_dummy::refresh_vap_info(int vap_id) { return true; }
 
 bool base_wlan_hal_dummy::refresh_vaps_info(int id) { return true; }
 
-bool base_wlan_hal_dummy::process_ext_events() { return true; }
+/**
+ * @brief process simulated events
+ *        events are expected to be simulated by writing the event
+ *        string to the first line in the EVENT file.
+ *        For example, simulating client connected event:
+ *        echo "STA_CONNECTED,11:22:33:44:55:66"
+ *
+ * @return true on success
+ * @return false on failure
+ */
+bool base_wlan_hal_dummy::process_ext_events()
+{
+    static char buffer[EVENT_BUF_LEN];
+
+    LOG(DEBUG) << "in process external events";
+
+    // dummy read to clear event
+    int length = read(m_fd_ext_events, buffer, EVENT_BUF_LEN);
+    if (length < 0) {
+        LOG(ERROR) << "Failed to read from m_fd_ext_events";
+        return false;
+    }
+
+    std::string event_file = std::string(BEEROCKS_TMP_PATH) + "/" + get_iface_name() + "/EVENT";
+    std::ifstream stream;
+    stream.open(event_file);
+    if (!stream.is_open()) {
+        LOG(DEBUG) << "Invalid event, missing " << event_file << " file";
+        return true;
+    }
+    std::string event;
+    std::getline(stream, event);
+    LOG(DEBUG) << "Received event " << event;
```

Comment:
I've shown an example that the event is processed, but couldn't see it here. The event is being printed and that it.
I guess you have pending changes you haven't committed yet, am I right?